RUMM-553 Fresco integration with Datadog SDK (#347)
DataDog/dd-sdk-android: dd-sdk-android-fresco/README.md

# Datadog Integration for Fresco

## Getting Started 

To include the Datadog integration for [Fresco][1] in your project, simply add the
following to your application's `build.gradle` file.

```
repositories {
    maven { url "https://dl.bintray.com/datadog/datadog-maven" }
}

dependencies {
    implementation "com.datadoghq:dd-sdk-android:<latest-version>"
    implementation "com.datadoghq:dd-sdk-android-fresco:<latest-version>"
}
```

### Initial Setup

Before you can use the SDK, you need to setup the library with your application
context and your API token. You can create a token from the Integrations > API
in Datadog. **Make sure you create a key of type `Client Token`.**

```kotlin
class SampleApplication : Application() {

    override fun onCreate() {
        super.onCreate()

>         val config = DatadogConfig.Builder(BuildConfig.DD_CLIENT_TOKEN).build()

Reviewer: I made the same mistake for Glide, we need to add the `envName` and `applicationId` in our sample code in our Readme. If they copy paste it as is, it won't compile.

Author: I will fix it also in Glide

        Datadog.initialize(this, config)

        val  logger = Logger.Builder()
                .setNetworkInfoEnabled(true)
                .setLogcatLogsEnabled(true)
                .setDatadogLogsEnabled(true)
                .build();
    }
}
```

> Following Fresco's [Generated API documentation][2], you then need to create your own `OkHttpImagePipelineConfigFactory` by providing your own `OkHttpClient` having the `DatadogInterceptor` attached,
> followed by providing the `DatadogFrescoCacheListener` through the `DiskCacheConfig`.

Reviewer: ```suggestion
Following Fresco's [Generated API documentation][2], you need to create your own `OkHttpImagePipelineConfigFactory` by providing your own `OkHttpClient` (configured with `DatadogInterceptor`). You can also add an instance of `DatadogFrescoCacheListener` in your `DiskCacheConfig`.
```

Doing so will automatically track Fresco's network requests (creating both APM Traces and RUM Resource events), and will also listen for disk cache errors (creating RUM Error events).

```kotlin
    val config = OkHttpImagePipelineConfigFactory.newBuilder(context, okHttpClient)
        .setMainDiskCacheConfig(
            DiskCacheConfig.newBuilder(context)
                .setCacheEventListener(DatadogFrescoCacheListener())
                .build()
        )
        .build()
    Fresco.initialize(context, config)
```

> That's it, now all your Fresco logs will be sent to Datadog automatically.

Reviewer: Not sure if this line is necessary.
